cluster/gce/manifests/pdcsi-controller.yaml

```diff
@@ -0,0 +1,165 @@
+    env:
+    - name: PDCSI_NAMESPACE
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.namespace
+    image: k8s.gcr.io/sig-storage/csi-attacher:v3.2.1
+    livenessProbe:
+      failureThreshold: 1
+      httpGet:
+        path: /healthz/leader-election
+        port: http-endpoint
+      initialDelaySeconds: 10
+      periodSeconds: 20
+      timeoutSeconds: 10
+    name: csi-attacher
+    ports:
+    - containerPort: 22012
+      name: http-endpoint
+      protocol: TCP
+    volumeMounts:
+    - mountPath: /csi
+      name: socket-dir
+    - mountPath: /etc/srv/kubernetes/pdcsi-controller
+      name: srv
+      readOnly: true
+  - args:
+    - --v=5
+    - --csi-address=/csi/csi.sock
+    - --http-endpoint=:22013
+    - --leader-election
+    - --leader-election-namespace=$(PDCSI_NAMESPACE)
+    - --handle-volume-inuse-error=false
+    - --kubeconfig=/etc/srv/kubernetes/pdcsi-controller/kubeconfig
+    env:
+    - name: PDCSI_NAMESPACE
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.namespace
+    image: k8s.gcr.io/sig-storage/csi-resizer:v1.2.0
+    livenessProbe:
+      failureThreshold: 1
+      httpGet:
+        path: /healthz/leader-election
+        port: http-endpoint
+      initialDelaySeconds: 10
+      periodSeconds: 20
+      timeoutSeconds: 10
+    name: csi-resizer
+    ports:
+    - containerPort: 22013
+      name: http-endpoint
+      protocol: TCP
+    volumeMounts:
+    - mountPath: /csi
+      name: socket-dir
+    - mountPath: /etc/srv/kubernetes/pdcsi-controller
+      name: srv
+      readOnly: true
+  - args:
+    - --v=5
+    - --csi-address=/csi/csi.sock
+    - --metrics-address=:22014
+    - --leader-election
+    - --leader-election-namespace=$(PDCSI_NAMESPACE)
+    - --timeout=300s
+    - --kubeconfig=/etc/srv/kubernetes/pdcsi-controller/kubeconfig
+    env:
+    - name: PDCSI_NAMESPACE
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.namespace
+    image: k8s.gcr.io/sig-storage/csi-snapshotter:v3.0.3
+    name: csi-snapshotter
+    volumeMounts:
+    - mountPath: /csi
+      name: socket-dir
+    - mountPath: /etc/srv/kubernetes/pdcsi-controller
+      name: srv
+      readOnly: true 
+  - args:
+    - --v=5
+    - --endpoint=unix:/csi/csi.sock
+    - --cloud-config=/etc/gce.conf
+    - --run-node-service=false
+    image: gke.gcr.io/gcp-compute-persistent-disk-csi-driver:v1.2.1-gke.0
+    name: gce-pd-driver
+    volumeMounts:
+    - mountPath: /csi
+      name: socket-dir
+    - mountPath: /etc/gce.conf
+      name: cloud-config-mount
+      readOnly: true
+  hostNetwork: true
+  nodeSelector:
+    kubernetes.io/os: linux
+  priorityClassName: system-node-critical
+  volumes:
+  - emptyDir: {}
+    name: socket-dir
+  - name: cloud-config-mount
+    hostPath:
+      path: /etc/gce.conf
+      type: FileOrCreate
+  - name: srv
+    hostPath:
+      path: /etc/srv/kubernetes/pdcsi-controller
```

Comment:
Is the main reason to deploy the manifest as a static pod vs normal pod because of credentials issues?

Previously, our test infra was able to collect driver logs from the tests. As a static pod, do we need to add additional logic to collect logs?

Reply:
Yes we were having trouble installing controller on master node with addon-manager.

Reply:
Yes, only the master node gets the cloud provider credentials.

We probably need logic to collect the logs. But we've also discovered that kubetest2 has broken the log collection for all the other master containers anyway (issue forthcoming).